synapse/lib/node.py

```diff
@@ -664,6 +664,12 @@ async def delete(self, force=False):
         for name in self.props.keys():
             edits.extend(await self._getPropDelEdits(name, init=True))
 
+        # Only remove nodedata if we're in a layer that doesn't have the full node
+        sode = await self.snap.view.layers[0].getStorNode(self.buid)
+        if sode.get('valu') is None:
```

Comment:
You should be able to use the bylayer info in the node for this like:
```suggestion
        if self.snap.wlyr.iden != self.bylayer['ndef']:
```